CaseTriggerHandler.cls

```diff
@@ -0,0 +1,108 @@
+/******************************************************************
+ * @Name         : CaseTriggerHandler
+ * @Description  : Methods of Contact Trigger
+ * @Created By   : Cagri Kilic
+ * @Created Date : Jan 18, 2023
+ * @Modification Log :
+ ******************************************************************
+ * Version        Developer        Date        Description
+ *------------------------------------------------------------
+ *
+ ******************************************************************/
+public with sharing class CaseTriggerHandler {
+    private static final String RECORD_TYPE_LABEL = 'Person Account';
+    private static final String BRAND_TYPE_1 = 'Option 1';
+    private static final String BRAND_TYPE_2 = 'Option 2';
+    private static final String ORIGIN_TYPE_1 = 'Phone';
+    private static final String ORIGIN_TYPE_2 = 'Email';
+    private static String recordTypeId = Schema.SObjectType.Account.getRecordTypeInfosByName().get(RECORD_TYPE_LABEL).getRecordTypeId();
+    /******************************************************************************
+     * @Name               : addAccountToCase
+     * @Description        : Adds Person Account to Case
+     * @Created By         : Cagri Kilic
+     * @Created Date       : Jan 18, 2023
+     * @Param newCases     : List<Case> - Case list
+     *****************************************************************************/
+    public static void addAccountToCase(List<Case> newCases) {
+        try {
+            List<Account> newAccounts = new List<Account>();
+            Map<String, List<String>> oldCasesMap = new Map<String, List<String>>();
+            List<Case> casesWithoutAccount = new List<Case>();
+            List<Case> oldCases = [
+                SELECT SuppliedEmail, Brand__c, Id, AccountId
+                FROM Case
+                WHERE SuppliedEmail != NULL
+            ];
+            for (Case c : oldCases) {
+                if (oldCasesMap.containsKey(c.SuppliedEmail)) {
+                    oldCasesMap.get(c.SuppliedEmail).add(c.Brand__c);
+                    oldCasesMap.get(c.SuppliedEmail).add(c.AccountId);
+                } else {
+                    oldCasesMap.put(c.SuppliedEmail, new List<String>{ c.Brand__c, c.AccountId });
```

Comment:
please create map of maps, like
Map<String, Map<String, Id>> = {'email1' => {'brand' => 'brand1', 'accountId' => 'accountId1'}